### ISSUE TITLE
Is there a Epoch 0.3.0 guide or tutorial? & Cmake error on Windows using MinGW as gcc

### ISSUE THREAD

**Issue by Jakio815** [lf-lang/lingua-franca#1543]
I'm curious if there is a Epoch guide page for "using" it not "installing".

Also I get an cmake error when pushing the gear button.

```
Generating code for: platform:/resource/RegressionTest/src/RegressionTest.lf
******** mode: EPOCH
******** generated sources: C:\Users\user\lf-workspace\RegressionTest\src-gen\RegressionTest
******** Using the NP runtime scheduler.
******** Using 1 threads to compile the program.
Cleaning C:\Users\user\lf-workspace\RegressionTest\src-gen\RegressionTest\build
--- Current working directory: C:\Users\user\lf-workspace\RegressionTest\src-gen\RegressionTest\build
--- Executing command: cmake -DCMAKE_INSTALL_PREFIX=C:/Users/user/lf-workspace/RegressionTest -DCMAKE_INSTALL_BINDIR=bin C:/Users/user/lf-workspace/RegressionTest/src-gen/RegressionTest -DCMAKE_SYSTEM_VERSION="10.0"
-- Building for: NMake Makefiles
-- Configuring incomplete, errors occurred!
See also "C:/Users/user/lf-workspace/RegressionTest/src-gen/RegressionTest/build/CMakeFiles/CMakeOutput.log".
CMake Error at CMakeLists.txt:2 (project):
  Running

   'nmake' '-?'

  failed with:

   The system cannot find the file specified


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
WARNING: CMake Error at CMakeLists.txt:2 (project):
  Running

   'nmake' '-?'

  failed with:

   The system cannot find the file specified


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
WARNING: C:\Users\user\lf-workspace\RegressionTest\src\RegressionTest.lf line 1
Warning in imported file: C:\Users\user\lf-workspace\RegressionTest\src\RegressionTest.lf
Compiled binary is in C:\Users\user\lf-workspace\RegressionTest\bin
```

My environment is somewhat complicated. I have a Windows machine, and I use WSL2 to use Linux. I usually develop inside Linux, but I have to use the Epoch 0.3.0 in Windows environment. 

I found out that there is a solution in https://stackoverflow.com/questions/69338088/error-while-configuring-cmake-project-running-nmake-failed adding `-G "MinGW Makefiles"` this flag. Also I found that there was a same issue https://github.com/lf-lang/lingua-franca/issues/1158

Can anyone help?

**Comment by erlingrj** [comment]
Regarding your second question. I was facing similar issue and after some trial and error I ended up installing the whole Microsoft Visual Studio suite. I could not find a way to just install the C compiler. 

Even on Windows, I prefer to install Lingua Franca on WSL (either download pre-built LFC or build from sources) and then use VS Codes ability to execute from WSL. You can either just install the VSCode extension for Lingua Franca and compile programs using VSCode commands, or (as I prefer) install Lingua Franca in WSL and compile programs from the command line.

**Comment by Jakio815** [comment]
@erlingrj Yeah I also prefer using WSL with either vscode extensions or command line compiling when I use it for myself. But I think the point of [epoch_ide_0.3.0-win32.win32.x86_64.zip]( https://github.com/lf-lang/lingua-franca/releases/tag/v0.3.0#:~:text=epoch_ide_0.3.0%2Dwin32.win32.x86_64.zip) is to use Epoch on Windows without Linux. Actually, I'm trying to use LF in class as a TA, and I think there would be some Windows only available users.

**Comment by Jakio815** [comment]
I think `set (CMAKE_GENERATOR "MinGW Makefiles" CACHE INTERNAL "" FORCE)` will be our solution according to [this](https://stackoverflow.com/questions/11269833/cmake-selecting-a-generator-within-cmakelists-txt).
I see that we have 'cmake-include' [properties](https://www.lf-lang.org/docs/handbook/target-declaration?target=c#files:~:text=the%20compilation%20process.-,cmake%2Dinclude,-target%20C%20%7B) to add, and I tried it on Epoch, but it didn't work. However, the problem is that according to [this](https://stackoverflow.com/questions/11269833/cmake-selecting-a-generator-within-cmakelists-txt#:~:text=in%20CMakeLists.txt%2C%20then%20in%20the%20very%20first%20run%20of%20cmake%20../source%20(without%20%2DG)%20the%20default%20generator%20is%20used.%20The%20variable%20CMAKE_GENERATOR%20is%20stored%20in%20the%20cache%2C%20though.%20So%20if%20you%20rerun%20cmake%20../source%20afterwards%2C%20it%20will%20use%20the%20generator%20as%20specified%20in%20the%20CMAKE_GENERATOR%20variable%20in%20the%20cache.) even adding  `set (CMAKE_GENERATOR "MinGW Makefiles" CACHE INTERNAL "" FORCE)` won't work unless the cmake is executed twice, and since every build it cleans the original generation, this won't happen. Are there any ideas about this?

**Comment by erlingrj** [comment]
Is Microsoft Visual Studio and MSVC not an option? https://visualstudio.microsoft.com/vs/features/cplusplus/

**Comment by hokeun** [comment]
@a-sr @lhstrh Do you think you could take a look at this issue? It looks like Windows users may have similar problems. Thank you!

**Comment by Jakio815** [comment]
@erlingrj  I found out that epoch Windows build uses MSVC as c compiler in default. I finally figured the `nmake -? ` part.
However, I still find errors that the CMAKE_C_COMPILER is not found.
```
Generating code for: platform:/resource/test/src/HelloWorld.lf
******** mode: EPOCH
******** generated sources: C:\Users\HYU\lf-workspace\test\src-gen\HelloWorld
******** Using the NP runtime scheduler.
******** Using 1 threads to compile the program.
Cleaning C:\Users\HYU\lf-workspace\test\src-gen\HelloWorld\build
--- Current working directory: C:\Users\HYU\lf-workspace\test\src-gen\HelloWorld\build
--- Executing command: cmake -DCMAKE_INSTALL_PREFIX=C:/Users/HYU/lf-workspace/test -DCMAKE_INSTALL_BINDIR=bin C:/Users/HYU/lf-workspace/test/src-gen/HelloWorld -DCMAKE_SYSTEM_VERSION="10.0"
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version  to target Windows 10.0.
-- The C compiler identification is unknown
-- Configuring incomplete, errors occurred!
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_C_COMPILER could be found.



WARNING: CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_C_COMPILER could be found.
WARNING: C:\Users\HYU\lf-workspace\test\src\HelloWorld.lf line 1
Warning in imported file: C:\Users\HYU\lf-workspace\test\src\HelloWorld.lf
Compiled binary is in C:\Users\HYU\lf-workspace\test\bin
```

I added `C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\bin\Hostx64\x64
` to Path, but it still doesn't work.

**Comment by lhstrh** [comment]
I would like to help, but I'm not a Windows user. Perhaps @erlingrj could help? I also just noticed that https://github.com/lf-lang/lingua-franca/issues/1158 has gone unaddressed, which seems related.

**Comment by erlingrj** [comment]
Unfortunately, I dont have a Windows machine anymore. My only advice is installing the complete MSVC toolchain, which is a rather big dependency. This should result in CMake getting correctly configured and `CMAKE_C_COMPILER` etc being set
